src/Document/Metadata/Metadata.php

```diff
@@ -86,6 +94,14 @@ private function loadAttributes(\ReflectionProperty|\ReflectionMethod $attribute
             if ($attribute instanceof SortableAttribute) {
                 $this->sortableAttributes[$name] = new Sortable($name, $attribute->direction);
             }
+
+            if ($attribute instanceof MapProductOption) {
+                $this->mapProductOptions[$name] = $attribute->codes;
```

Comment:
Should we validate here that the type of the property is `array`? Same question for the `MapProductAttribute`